core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -77,22 +76,18 @@ public static PrepareContractionHierarchies fromGraphHopperStorage(GraphHopperSt
     }
 
     private PrepareContractionHierarchies(GraphHopperStorage ghStorage, CHConfig chConfig) {
+        graph = ghStorage;
         chGraph = ghStorage.getCHGraph(chConfig.getName());
         if (chGraph == null)
             throw new IllegalArgumentException("There is no CH graph '" + chConfig.getName() + "', existing: " + ghStorage.getCHGraphNames());
         this.chConfig = chConfig;
         params = Params.forTraversalMode(chConfig.getTraversalMode());
-        updatedNeighbors = new IntHashSet(50);
+        nodes = chGraph.getNodes();
         if (chConfig.getTraversalMode().isEdgeBased()) {
             TurnCostStorage turnCostStorage = chGraph.getBaseGraph().getTurnCostStorage();
             if (turnCostStorage == null) {
                 throw new IllegalArgumentException("For edge-based CH you need a turn cost storage");
             }
-            prepareGraph = PrepareCHGraph.edgeBased(chGraph, chConfig.getWeighting());
-            nodeContractor = new EdgeBasedNodeContractor(prepareGraph, pMap);
-        } else {
-            prepareGraph = PrepareCHGraph.nodeBased(chGraph, chConfig.getWeighting());
-            nodeContractor = new NodeBasedNodeContractor(prepareGraph, pMap);
```

Comment:
Here we were allocating quite a bit of memory even when re-starting the server and just loading the preparations (no new preparation). We should really make PrepareContractionHierarchies about preparing CH and not use it as a 'handle' to the preparations available on the server...